### ISSUE TITLE
Materialize select creating infinite span.caret on update

### ISSUE THREAD

**Issue by ahlechandre** [Dogfalo/materialize#1876]
trying to make select disabled dinamically by switch checkbox
```javascript
    var
      hasItem = document.querySelector('input[name="has-item"]'), // materialize switch
      item = document.querySelector('select[name="item"]');

    $( item ).material_select();

    hasItem.addEventListener( 'change', function ()
    {
      $( item ).material_select('destroy');
      if ( this.checked === true )
      {
        if ( item.hasAttribute( 'disabled' ) )
        {
          item.removeAttribute( 'disabled' );
        }
      }
      else
      {
        if ( !item.hasAttribute( 'disabled' ) )
        {
          item.setAttribute( 'disabled', '' );
        }
      }
     $( item ).material_select();
    } );
```
after some switch actions on/off
![materialize_select](https://cloud.githubusercontent.com/assets/7119058/9151933/2c204d5a-3dec-11e5-8be4-e74c1cddd134.jpg)

can someone explain to me why this occurs?

**Comment by davismj** [comment]
+1

**Comment by davismj** [comment]
In my 0.97.0 copy, I have the following lines starting from 2730:
```
// Tear down structure if Select needs to be rebuilt
      var lastID = $select.data('select-id');
      if (lastID) {
        $select.parent().find('i').remove();
        $select.parent().find('input').remove();
```

This does not match the [latest code](https://github.com/Dogfalo/materialize/blob/master/js/forms.js#L284). So it seems this issue is fixed and will be resolved when a new materialize is released.

**Comment by carlosperate** [comment]
This issue has been solved after the 0.97.0 release. If you can't wait until the next release you could start using the files from https://github.com/Dogfalo/materialize/tree/master/bin.

**Comment by ahlechandre** [comment]
thanks
